warn/scrapers/hi.py

```diff
@@ -0,0 +1,89 @@
+from pathlib import Path
+
+from bs4 import BeautifulSoup
+
+from .. import utils
+
+__authors__ = ["Ash1R"]
+__tags__ = ["html", "pdf"]
+__source__ = {
+    "name": "Workforce Development Hawaii",
+    "url": "https://labor.hawaii.gov/wdc/real-time-warn-updates/",
+}
+
+
+def scrape(
+    data_dir: Path = utils.WARN_DATA_DIR,
+    cache_dir: Path = utils.WARN_CACHE_DIR,
+) -> Path:
+    """
+    Scrape data from Hawaii.
+
+    Keyword arguments:
+    data_dir -- the Path were the result will be saved (default WARN_DATA_DIR)
+    cache_dir -- the Path where results can be cached (default WARN_CACHE_DIR)
+    Returns: the Path where the file is written
+    """
+    page = utils.get_url("https://labor.hawaii.gov/wdc/real-time-warn-updates/")
+    soup = BeautifulSoup(page.text)
+    tags = soup.select("p a[href*=pdf]")
+    p_tags = [i.parent.get_text().replace("\xa0", " ").split("\n") for i in tags]
+    clean_p_tags = [j for i in p_tags for j in i]
+
+    dates = [k.split("–")[0] for k in clean_p_tags]
+    headers = ["Company", "Date", "PDF url"]
+    data = [headers]
+    for i in range(len(dates)):
+        if len(dates[i]) > 20:
+            includes_date = False
+            for j in [
+                "January",
+                "February",
+                "March",
+                "April",
+                "May",
+                "June",
+                "July",
+                "August",
+                "September",
+                "October",
+                "November",
+                "December",
+            ]:
+                if j in dates[i].split(" ")[0]:
+                    dates[i] = (
+                        dates[i].split(" ")[0]
+                        + " "
+                        + dates[i].split(" ")[1]
+                        + " "
+                        + dates[i].split(" ")[2]
+                    )
+                    includes_date = True
+                elif j in dates[i].split(" ")[-3]:
+                    dates[i] = (
+                        dates[i].split(" ")[-3]
+                        + " "
+                        + dates[i].split(" ")[-2]
+                        + " "
+                        + dates[i].split(" ")[-1]
+                    )
```

Comment:
Can you explain to me why we need all this date parsing? Is there a reason we can't pass the dates into [strptime](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior) or [dateutil](https://pypi.org/project/python-dateutil/) or another date parser?

Reply:
Some entries have dates, but other entries are corrections which don't have dates. 
Some entries have the date at the end, and others at the beginning, hence the if and elif near the bottom. I extracted the parts of the entry which contain the date and put them in a string. 

Can dateutil or strptime read the whole entry and extract the date?

Reply:
I think it might. Give it a try.

Reply:
Gotta love fuzzy parsing, it works. 
However, 
<img width="729" alt="Screen Shot 2023-01-18 at 12 51 05 AM" src="https://user-images.githubusercontent.com/69091220/213126031-a1d1db3b-ff93-437b-b8e5-d2b614da0d5e.png">
When I try to commit, it says that "library stubs" are not installed for dateutil or dateutil.parser. I tried running the suggested commands, but I'm still getting the same error.

Reply:
Try running adding `types-python-dateutil` here: https://github.com/biglocalnews/warn-scraper/blob/main/.pre-commit-config.yaml#L51-L53

Reply:
The error is gone and the code is pushed, thanks!